relay-server/src/extractors/signed_json.rs

```diff
@@ -101,7 +100,36 @@ where
             .ok_or(SignatureError::UnknownRelay)?;
 
         let body = Bytes::from_request(request, state).await?;
-        let inner = relay.public_key.unpack(&body, &signature, None)?;
+        if !relay.public_key.verify(&body, &signature) {
+            Err(SignatureError::BadSignature(UnpackError::BadSignature))
+        } else {
+            Ok(SignedBytes { body, relay })
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct SignedJson<T> {
+    pub inner: T,
+    pub relay: RelayInfo,
+}
+
+#[axum::async_trait]
+impl<T, B> FromRequest<ServiceState, B> for SignedJson<T>
+where
+    T: DeserializeOwned,
+    B: axum::body::HttpBody + Send + 'static,
+    B::Data: Send,
+    B::Error: Into<axum::BoxError>,
+{
+    type Rejection = SignatureError;
+
+    async fn from_request(
+        request: Request<B>,
+        state: &ServiceState,
+    ) -> Result<Self, Self::Rejection> {
+        let SignedBytes { body, relay } = SignedBytes::from_request(request, state).await?;
```

Comment:
What do we need `SignedJson` for? We have `SignedBytes` and use it here, plus the actor gets bytes and assumes it's valid JSON.

Reply:
`SignedJson` is used by all other endpoints, which we're not refactoring in this PR. The usual pattern is to validate + deserialize JSON directly in the extractor.